library/src/image.cxx

```diff
@@ -351,6 +353,145 @@ std::vector<unsigned char> image::saveBuffer(SaveFormat format) const
   }
 }
 
+//----------------------------------------------------------------------------
+std::string image::toTerminalText() const
+{
+  const int depth = this->getChannelCount();
+  if (this->getChannelType() != ChannelType::BYTE || depth < 3 || depth > 4)
+  {
+    throw std::invalid_argument("image must be byte RGB or RGBA");
+  }
+
+  int dims[3];
+  this->Internals->Image->GetDimensions(dims);
+  const int width = dims[0];
+  const int height = dims[1];
+  const unsigned char* content = static_cast<unsigned char*>(this->getContent());
+
+  /* Function to retrieve pixels so we can return:
+    - transparent black values for out-of-bounds coords,
+    - opaque alpha value when the image has no alpha channel.
+    Rendering with half blocks means 1 line of text represents 2 rows of pixels
+    so we _will_ attempt to access a line past the bottom if the height is not even.
+  */
+  const auto getPixel = [&content, width, height, depth](int x, int y, unsigned char* alpha)
+  {
+    if (x >= 0 && x < width && y >= 0 && y < height)
+    {
+      const size_t i = depth * ((height - 1 - y) * width + x);
+      *alpha = depth > 3 ? content[i + 3] : (unsigned char)255;
```

Comment:
Instead of casting a int, just use 0xff

Reply:
isn't `0xff` the same int tho?

Reply:
It is, but you won't have to cast it to `unsigned char` since it's typed to that already.